commit 11ad033e1c09c8b8e7bbaa72420f41ab8bcf0f63
Author: Flavio Cruz <flaviocruz@gmail.com>
Date:   Tue Jul 30 00:51:20 2024 -0400

    x86_64 hurd: ensure we have a large enough buffer to receive exception_raise requests.
    
    Message-ID: <gtxd6s4s7fi7hdrlb7zayq3akij7x6jqawwq3zfl3v4nqspulo@euucuzeonrl6>

diff --git a/hurd/hurdfault.c b/hurd/hurdfault.c
index 5110c6030f..1fe973f54d 100644
--- a/hurd/hurdfault.c
+++ b/hurd/hurdfault.c
@@ -121,7 +121,14 @@ faulted (void)
   struct
     {
       mach_msg_header_t head;
-      char buf[64];
+        /* This is the size of the exception_raise request
+         * including mach_msg_header_t.
+         * See generated code in faultexc_server.c.  */
+#ifdef __LP64__
+        char buf[112];
+#else
+        char buf[64];
+#endif
     } request;
   mig_reply_header_t reply;
   extern int _hurdsig_fault_exc_server (mach_msg_header_t *,
